app/services/books/index.rb

```diff
@@ -52,8 +58,22 @@ def exists?
       OsElasticsearchClient.instance.indices.exists?(index: name)
     end
 
+    def not_exists?
+      !OsElasticsearchClient.instance.indices.exists?(index: name)
+    end
+
     private
 
+    def wait_until(this_happens)
+      tries = 1
+      until self.send(this_happens) || tries > 5  do
+        Rails.logger.debug("Waiting for #{name} 1 sec for #{name} to #{this_happens.to_s}, tries: #{tries}")
+        sleep(2)
+        tries += 1
```

Comment:
Is 10 seconds a long enough timeout?  And if we do hit the timeout should we raise an exception?  Otherwise it'll just explode later (at least our code will)

Reply:
yeah, let's raise it to 1 minute and raise an exception (a minute would justify an exception)